lib/install/mutate-into-logical-tree.js

```diff
@@ -70,18 +70,27 @@ function translateTree_ (tree, seen) {
   tree.children.forEach(function (child) {
     pkg.dependencies[moduleName(child)] = translateTree_(child, seen)
   })
-  Object.keys(tree.missingDeps).forEach(function (name) {
+
+  function markMissing (name, requiredBy) {
     if (pkg.dependencies[name]) {
+      if (pkg.dependencies[name].missing) return
       pkg.dependencies[name].invalid = true
```

Comment:
Are the `invalid`, `realName`, `extraneous`, and `missing` values documented anywhere? More generally, is there some kind of schema for what properties can be attached to a package in the logical tree? I did a fair amount of searching even to discover how `translateTree` is meant to be used, and even now am a little confused as to what distinction to draw between `invalid` and `extraneous` or `missing`.